cmd/release.go

```diff
@@ -67,31 +72,122 @@ func releaseRootCommand(cmd *cobra.Command, _ []string) {
+	}
 
-	if !specific && rs.Status == statusFail {
-		qwm(ExitCodeError, "project does not pass release criteria")
+	iac, err := cmd.Flags().GetBool("iac")
+	if err != nil {
+		qwm(ExitCodeError, "failed to parse iac flag")
 	}
 
-	if sast && rs.SAST.Status == statusFail {
-		qwm(ExitCodeError, "project does not pass SAST release criteria")
+	isSpecific := sast || dast || pentest || iast || sca || cs || iac
+
+	var spesificMap = make(map[string]bool, 0)
+	spesificMap["SAST"] = sast
+	spesificMap["DAST"] = dast
+	spesificMap["PENTEST"] = pentest
+	spesificMap["IAST"] = iast
+	spesificMap["SCA"] = sca
+	spesificMap["CS"] = cs
+	spesificMap["IAC"] = iac
+
+	isReleaseFailed(rs, isSpecific, spesificMap)
+}
+
+func isReleaseFailed(release *client.ReleaseStatus, isSpecific bool, specificMap map[string]bool) {
+	const statusFail = "fail"
+
+	if release.Status != statusFail {
+		return
+	}
+
+	var failedScans = make(map[string]string, 0)
+
+	if release.SAST.Status == statusFail {
+		failedScans["SAST"] = release.SAST.ScanID
+	}
+	if release.DAST.Status == statusFail {
+		failedScans["DAST"] = release.DAST.ScanID
+	}
+	if release.PENTEST.Status == statusFail {
+		failedScans["PENTEST"] = release.PENTEST.ScanID
+	}
+	if release.IAST.Status == statusFail {
+		failedScans["IAST"] = release.IAST.ScanID
+	}
+	if release.SCA.Status == statusFail {
+		failedScans["SCA"] = release.SCA.ScanID
+	}
+	if release.CS.Status == statusFail {
+		failedScans["CS"] = release.CS.ScanID
+	}
+	if release.IAC.Status == statusFail {
+		failedScans["IAC"] = release.IAC.ScanID
 	}
 
-	if dast && rs.DAST.Status == statusFail {
-		qwm(ExitCodeError, "project does not pass DAST release criteria")
+	if verbose {
+		c, err := client.New()
+		if err != nil {
+			qwe(ExitCodeError, err, "could not initialize Kondukto client")
+		}
+
+		for toolType, scanID := range failedScans {
+			if isSpecific {
+				if !specificMap[toolType] {
+					continue
+				}
+			}
+
+			fmt.Println()
+			fmt.Println("-----------------------------------------------------------------")
+			fmt.Printf("[!] project does not pass release criteria due to [%s] failure\n", toolType)
+			scan, err := c.FindScanByID(scanID)
+			if err != nil {
+				qwe(ExitCodeError, err, "failed to fetch scan summary")
+			}
+
+			printScanSummary(scan)
+			fmt.Println("-----------------------------------------------------------------")
+		}
 	}
 
-	if sca && rs.SCA.Status == statusFail {
-		qwm(ExitCodeError, "project does not pass SCA release criteria")
+	var failedToolTypes []string
+
+	for toolType := range failedScans {
+		if isSpecific {
+			if specificMap[toolType] {
+				failedToolTypes = append(failedToolTypes, toolType)
+			}
+		} else {
+			failedToolTypes = append(failedToolTypes, toolType)
+		}
 	}
 
-	qwm(ExitCodeSuccess, "project passes release criteria")
+	if len(failedToolTypes) == 0 {
+		returnMSG := fmt.Sprintf("project passes release criteria")
+		qwe(ExitCodeSuccess, errors.New(returnMSG))
+	} else {
```

Comment:
You do not need these changes to determine if the result was successful or not. 
There is a `Status` field on the root, 
```go
	PlaybookResults struct {
		Status  string             `json:"status" bson:"status"`
		SAST    PlaybookTypeDetail `json:"sast" bson:"sast"`
		DAST    PlaybookTypeDetail `json:"dast" bson:"dast"`
		PENTEST PlaybookTypeDetail `json:"pentest" bson:"pentest"`
		IAST    PlaybookTypeDetail `json:"iast" bson:"iast"`
		SCA     PlaybookTypeDetail `json:"sca" bson:"sca"`
		CS      PlaybookTypeDetail `json:"cs" bson:"cs"`
		IAC     PlaybookTypeDetail `json:"iac" bson:"iac"`
	}
```
It holds the overall status.

Reply:
Actually you right but we can use kdt like this:
`kdt release -p Juiceshop --sast --cs`

With this situation, kdt checks only `sast` and `cs` types. (Not all)
By the way, kdt must check failed types that only this types.

If you use kdt like this:
`kdt release -p Juiceshop` (With not specific types)
Kdt will check all types.
So, we must log all failed types.